lib/DocumentFilter/DocumentFilterField.test.js

```diff
@@ -1,96 +1,98 @@
-            initialValues={{
-              filters: initalFilters
-            }}
-            onSubmit={onSubmit}
-          >
-            <FieldArray name="filters">
-              {({ fields }) => fields.map((name, index) => (
-                <DocumentFilterField
-                  key={`${name}[${index}]`}
-                  categoryValues={categoryValues}
-                  filterName="docs"
-                  index={index}
-                  name={name}
-                />
-              ))}
-            </FieldArray>
-          </TestForm>
-          ,
-        </MemoryRouter>,
-        translationsProperties
-      );
-    });
+      expectedFields: 0,
+    },
+  ])(
+    'Render DocumentFilterField with $expectedFields in the array',
+    ({ initalFilters, expectedFields }) => {
+      beforeEach(() => {
+        renderComponent = renderWithIntl(
+          <MemoryRouter>
+            <TestForm
+              initialValues={{
+                filters: initalFilters,
+              }}
+              onSubmit={onSubmit}
+            >
+              <FieldArray name="filters">
+                {({ fields }) => fields.map((name, index) => (
+                  <DocumentFilterField
+                    key={`${name}[${index}]`}
+                    categoryValues={categoryValues}
+                    filterName="docs"
+                    index={index}
+                    name={name}
+                  />
+                ))
+                }
+              </FieldArray>
+            </TestForm>
+            ,
+          </MemoryRouter>,
+          translationsProperties
+        );
+      });
 
-    it('displays attibute label(s)', () => {
-      const { queryAllByText } = renderComponent;
-      expect(queryAllByText('Attribute')).toHaveLength(expectedFields);
-    });
+      it('displays attribute label(s)', () => {
+        const { queryAllByText } = renderComponent;
+        expect(queryAllByText('Attribute')).toHaveLength(expectedFields);
+      });
 
-    it('displays operator label(s)', () => {
-      const { queryAllByText } = renderComponent;
-      expect(queryAllByText('Operator')).toHaveLength(expectedFields);
-    });
+      it('displays operator label(s)', () => {
+        const { queryAllByText } = renderComponent;
+        expect(queryAllByText('Operator')).toHaveLength(expectedFields);
+      });
 
-    it('displays value label(s)', () => {
-      const { queryAllByText } = renderComponent;
-      expect(queryAllByText('Value')).toHaveLength(expectedFields);
-    });
+      it('displays value label(s)', () => {
+        const { queryAllByText } = renderComponent;
+        expect(queryAllByText('Value')).toHaveLength(expectedFields);
+      });
 
-    if (expectedFields > 0) {
-      test('renders the add rule button', async () => {
-        await Button('Add rule').exists();
+      it('renders DocumentFilterRule component with the correct number of times by clicking the "Add rule" button', async () => {
```

Comment:
This is a little overcomplicated. The big if/else isn't needed here. We can instead test:

- Does the right number of DocumentFilterRule exist? (Use queryAllByText so the 0 case works)
- Does add rule exist
- Describe clicking add rule
  - Has number of DocumentFilterRules gone up by one?

Reply:
In general the point of the `.each` at the top is so that this kind of case by case logic _isn't_ needed inside the tests themselves. Let me know if you wan tme to walk through this case with you cos it's an interesting shift in test writing